src/node/convolver.rs

```diff
@@ -0,0 +1,568 @@
+use super::{AudioNode, ChannelConfig, ChannelConfigOptions, ChannelInterpretation};
+use crate::buffer::AudioBuffer;
+use crate::context::{AudioContextRegistration, BaseAudioContext};
+use crate::render::{AudioParamValues, AudioProcessor, AudioRenderQuantum, RenderScope};
+use crate::RENDER_QUANTUM_SIZE;
+
+use crossbeam_channel::{Receiver, Sender};
+use realfft::{num_complex::Complex, RealFftPlanner};
+use std::sync::Mutex;
+
+/// Scale buffer by an equal-power normalization
+fn normalization(buffer: &AudioBuffer) -> f32 {
+    let gain_calibration = 0.00125;
+    let gain_calibration_sample_rate = 44100.;
+    let min_power = 0.000125;
+
+    // Normalize by RMS power.
+    let number_of_channels = buffer.number_of_channels();
+    let length = buffer.length();
+    let sample_rate = buffer.sample_rate();
+
+    let mut power: f32 = buffer
+        .channels()
+        .iter()
+        .map(|c| c.as_slice().iter().map(|&s| s * s).sum::<f32>())
+        .sum();
+
+    power = (power / (number_of_channels * length) as f32).sqrt();
+
+    // Protect against accidental overload.
+    if !power.is_finite() || power.is_nan() || power < min_power {
+        power = min_power;
+    }
+
+    let mut scale = 1. / power;
+
+    // Calibrate to make perceived volume same as unprocessed.
+    scale *= gain_calibration;
+
+    // Scale depends on sample-rate.
+    scale *= gain_calibration_sample_rate / sample_rate;
+
+    // True-stereo compensation.
+    if number_of_channels == 4 {
```

Comment:
I've saw that's in the spec but it seems weird...

Reply:
Looking at the diagram it makes sense: https://www.w3.org/TR/webaudio/#Convolution-channel-configurations
The output is stereo, so pairs of channels are summed, so compensate with 0.5.
I am more confused by `    let gain_calibration = 0.00125;` which comes out of nowhere. But hey, let's conform to the spec :)

Reply:
Yup this algorithm is quite strange, don't really get the idea with the hardcoded sample rate too... but as you say "let's conform to the spec :)"